src/middlewares/permissions.js

```diff
@@ -1,15 +1,27 @@
 const PermissionsChecker = require('../services/permissions-checker');
 const httpError = require('http-errors');
 const logger = require('../services/logger');
+const ConfigStore = require('../services/config-store');
 
 const getRenderingIdFromUser = user => user.renderingId;
 
-function createCheckPermission(environmentSecret, collectionName) {
-  function checkPermission(permissionName) {
+class PermissionMiddleWareCreator {
```

Comment:
[Wording] I kinda have an issue with the name `PermissionMiddleWareCreator` why wouldn't `PermissionsMiddleWare` be sufficient ?

Reply:
Because it's not a middleware it's something that creates middleware. When just reading the code you might want to juste use `PermissionMiddleware` in an `app.use(PermissionMiddleware)`. Which is wrong